R/pit.R

```diff
@@ -52,105 +52,74 @@
-#' @param full_output return all individual p_values and computed u_t values
-#' for the randomised PIT. Usually not needed.
-#' @param verbose if `TRUE` (default is `FALSE`) more error messages are printed.
-#' Usually, this should not be needed, but may help with debugging.
-#' @return a list with the following components:
-#' \itemize{
-#' \item `p_value`: p-value of the Anderson-Darling test on the
-#' PIT values. In case of integer forecasts, this will be the mean p_value
-#' from the `n_replicates` replicates
-#' \item `sd`: standard deviation of the p_value returned. In case of
-#' continuous forecasts, this will be NA as there is only one p_value returned.
-#' \item `hist_PIT` a plot object with the PIT histogram. Only returned
-#' if `plot = TRUE`. Call `plot(PIT(...)$hist_PIT)` to display the histogram.
-#' \item `p_values`: all p_values generated from the Anderson-Darling tests
-#' on the (randomised) PIT. Only returned if `full_output = TRUE`
-#' \item `u`: the u_t values internally computed. Only returned if
-#' `full_output = TRUE`
-#' }
-#' @importFrom goftest ad.test
-#' @importFrom stats runif sd
+#' @param n_replicates the number of draws for the randomised PIT for
+#' integer predictions.
+#' @inheritParams ae_median_sample
+#' @return A vector with PIT-values. For continuous forecasts, the vector will
+#' correspond to the length of `true_values`. For integer forecasts, a
+#' randomised PIT will be returned of length
+#' `length(true_values) * n_replicates`
+#' @seealso [pit()]
+#' @importFrom stats runif
 #' @examples
-#'
 #' ## continuous predictions
 #' true_values <- rnorm(30, mean = 1:30)
 #' predictions <- replicate(200, rnorm(n = 30, mean = 1:30))
-#' pit(true_values, predictions)
+#' pit <- pit_sample(true_values, predictions)
+#' plot_pit(pit)
 #'
 #' ## integer predictions
 #' true_values <- rpois(100, lambda = 1:100)
 #' predictions <- replicate(5000, rpois(n = 100, lambda = 1:100))
-#' pit(true_values, predictions,  n_replicates = 5)
-#'
+#' pit <- pit_sample(true_values, predictions, n_replicates = 50)
+#' plot_pit(pit)
 #' @export
 #' @references
 #' Sebastian Funk, Anton Camacho, Adam J. Kucharski, Rachel Lowe,
 #' Rosalind M. Eggo, W. John Edmunds (2019) Assessing the performance of
 #' real-time epidemic forecasts: A case study of Ebola in the Western Area
 #' region of Sierra Leone, 2014-15, <doi:10.1371/journal.pcbi.1006785>
+#' @keywords metric
 
-
-pit <- function(true_values,
-                predictions,
-                plot = TRUE,
-                full_output = FALSE,
-                n_replicates = 50,
-                num_bins = NULL,
-                verbose = FALSE) {
-
-
+pit_sample <- function(true_values,
+                       predictions,
+                       n_replicates = 100) {
 
   # error handling--------------------------------------------------------------
   # check al arguments are provided
   if (!all(c(methods::hasArg("true_values"), methods::hasArg("predictions")))) {
```

Comment:
Again, use `missing()` instead?

Reply:
so using `missing()` instead of `hasArg()`?

Reply:
done